mautrix_signal/commands/signal.py

```diff
@@ -287,3 +288,81 @@ async def raw(evt: CommandEvent) -> None:
             await evt.reply(
                 f"Got reply `{resp_type}`:\n\n```json\n{json.dumps(resp_data, indent=2)}\n```"
             )
+
+
+@command_handler(
+    needs_auth=True,
+    management_only=False,
+    help_section=SECTION_SIGNAL,
+    help_text=("Create a Signal group for the current Matrix room. "),
+)
+async def create(evt: CommandEvent) -> EventID:
+    if await po.Portal.get_by_mxid(evt.room_id):
+        return await evt.reply("This is already a portal room.")
+
+    title, about, levels, encrypted, avatar_url = await get_initial_state(
+        evt.az.intent, evt.room_id
+    )
+
+    portal = po.Portal(
+        chat_id=GroupID(""),
+        mxid=evt.room_id,
+        name=title,
+        topic=about,
+        encrypted=encrypted,
+        receiver="",
+        avatar_url=avatar_url,
+    )
+
+    if levels.get_user_level(evt.az.bot_mxid) < 51:
+        await evt.reply(
+            "Warning: The bot does not have privileges to demote moderators on Matrix. "
+            "Demotions from ADMINISTRATOR to DEFAULT will not be bridged unless you give "
+            f"redaction permissions to [{evt.az.bot_mxid}](https://matrix.to/#/{evt.az.bot_mxid})"
+        )
```

Comment:
Might be a good idea to check that the bot has a high enough level to change power levels in general, it might require PL 100

(also the message says redaction permissions)